tests/test_bug_regressions.py

```
@@ -0,0 +1,367 @@
+    var = np.abs(np.random.randn(channels).astype(dtype))
+
+    return gamma, beta, mean, var
+
+
+class TestBatchNormBackward(unittest.TestCase):
+    """4D backward divided by N instead of N*H*W, and eval-mode backward used stale stats."""
+
+    def test_backward_4d(self):
+        x = np.random.randn(3, 2, 4, 5)
+        gamma, beta, mean, var = get_bn_params(2)
+        f = lambda x_in: funcs.batch_normalization(x_in, gamma, beta, mean, var)
```

Ruff (0.15.15)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `f` as a `def`

(E731)